Use larger modal for add/edit dashboard cards

cconard96 opened the pull request:
## Checklist before requesting a review

- [x] I have read the CONTRIBUTING document.
- [x] I have performed a self-review of my code.

## Description

Fixes #19273
Use `modal-xl` for the modals for adding/editing dashboard cards so the UI isn't so cramped.

## Screenshots

Ignore the obvious widget image issues in dark mode.
Before             |  After
:-------------------------:|:-------------------------:
![Selection_523](https://github.com/user-attachments/assets/e9c8949a-4de2-46e9-9c8a-cbd92e33e56f) | ![Selection_524](https://github.com/user-attachments/assets/bf3560ff-7f15-4960-afd4-2c57a7b28881)

cconard96 commented:
![Selection_526](https://github.com/user-attachments/assets/f2f44a30-71c7-4773-8474-1a326fb8d7d0)

orthagh commented:
If we increase the size for the label, I think we will have a better proportions for the `-lg` breakpoint.
standard size is indeed too small, and xl is too much

cconard96 commented:
> (Did you delete your last comment @cconard96, or is there a bug in this GitHub issue display)

Deleted. I posted that the width of the labels and fields when "full_width" is set was hard-coded in the SCSS now, which would make simply chaning the label.field classes to adjust the width difficult. I deleted a while later because I decided I was just going to try to adapt the CSS to account for modal displays.
